php-checks/src/test/resources/checks/regex/StringReplaceCheck.php

```diff
@@ -56,4 +56,27 @@ function patternAsVariable() {
   //^^^^^^^^^^^^
   }
 
+  function pregReplaceCallAreOkIfLimitIsUsed() {
+    $init = "Bob is a Bird... Bob is a Plane... Bob is Superman!";
+
+    preg_replace("/\n/", " ", "source string", 2);
+
+    // test when fourth param is a variable
+    $varIndex=3;
+    preg_replace("/\n/", " ", "source string", $varIndex);
+    preg_replace("/\n/", " ", "source string", -$varIndex);
+
+
+    // test with named parameters
+    preg_replace(pattern: "/\n/", replacement: " ", limit: 2, subject: $init);
+
+    // as -1 is the limit default value, we should report these ones, even if the limit parameter is set
+    preg_replace("/\n/", " ", "source string", -1); // Noncompliant
+    preg_replace("/\n/", " ", "source string", -1, $count); // Noncompliant
+
+    // when count parameter is set, if limit is not, we should report as replacement with str_replace is possible
+    preg_replace(pattern: "/\n/", replacement: " ", subject: $init, count: $count); // Noncompliant
+
+  }
```

Comment:
I would get rid of the `pregReplaceCallAreOkIfLimitIsUsed `and I would keep only ` line#62` and `line#71` as a `compliant `example and move it to the already existing bunch of compliant functions above (you can add a comment if you want saying that it is `compliant `because the limit is set).

Similarly, I would keep `line#74` in the `noncompliant `functions above (you can add a comment if you want, saying that it is `noncompliant ` because the limit is set to the default value).

I think that the other tests do not add something in regards to your implementation, as for us, it is not important if the `count` parameter is set or not, and similarly is not so important if you have a number or a parameter, as we check if there is or not a minus at the start.
What do you think?

Reply:
You are completely right, the case where useful during implementation to ensure It work in all situation, but are no more needed now.